src/identity.rs

```diff
@@ -237,14 +240,10 @@ impl Password {
     /// Scope authentication to the given project.
     ///
     /// This is required in the most cases.
-    pub fn set_project_scope<S1, S2>(&mut self, project_name: S1, project_domain_name: S2)
-    where
-        S1: Into<String>,
-        S2: Into<String>,
-    {
+    pub fn set_project_scope(&mut self, project: IdOrName, domain: impl Into<Option<IdOrName>>) {
```

Comment:
What are we gaining from using `Into` here? Previously `Into` was used to be able to accept both `String` and `&str`, but I don't think we can convert anything into an `Option<IdOrName>`?

Reply:
A related thought: maybe we should make `IdOrName` `From<T> where T: Into<String>` by assuming a name? That would avoid a breaking change.

Reply:
> What are we gaining from using `Into` here? Previously `Into` was used to be able to accept both `String` and `&str`, but I don't think we can convert anything into an `Option<IdOrName>`?

The benefit of using `Into<IdOrName>` is that we do not have to wrap a value in `Some()`. This is because `From<T> for Option<T>` exists. This is just a nice convenience which means that one can either give a `IdOrName` directly or a `None`.



> A related thought: maybe we should make `IdOrName` `From<T> where T: Into<String>` by assuming a name? That would avoid a breaking change.

Personally I think this would make the interface difficult to understand because an id is also a `String`. I understand that using `IdOrName` is a bit of a pain, but I think the benefits of a clear, verbose interface is better than having a implicity conversion into name.